File: FFLabel/FFLabelObjcDemo/FFLabelObjcDemo/ViewController.m
//
//  ViewController.m
//  FFLabelObjcDemo
//
//  Created by 刘凡 on 15/7/22.
//  Copyright © 2015年 joyios. All rights reserved.
//

#import "ViewController.h"
#import "FFLabelObjcDemo-Swift.h"

@interface ViewController () <FFLabelDelegate>
@property (weak, nonatomic) IBOutlet FFLabel *label;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    FFLabel *s = [[FFLabel alloc] initWithFrame:CGRectZero];
    
    [self.view addSubview:s];
    
    s.delegate = self;
    s.text = @"#FFLabel#This is a @FFLabel Demo, access http://github.com/liufan321/fflabel can get the demo project. Follow @liufan2000 to get more information.";
    
    [s setPreferredMaxLayoutWidth:[UIScreen mainScreen].bounds.size.width - 20];
    s.translatesAutoresizingMaskIntoConstraints = NO;
    
    [self.view addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-10-[s]-10-|" options:0 metrics:nil views:@{@"s": s}]];
    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:s attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeTop multiplier:1.0 constant:100]];
    s.numberOfLines = 0;
}


- (void)labelDidSelectedLinkText:(FFLabel * __nonnull)label text:(NSString * __nonnull)text {
    NSLog(@"%@", text);
}

@end

File: FFLabel/FFLabel Swift Demo/FFLabel Swift Demo/ViewController.swift
//
//  ViewController.swift
//  FFLabel Swift Demo
//
//  Created by 刘凡 on 15/7/18.
//  Copyright © 2015年 joyios. All rights reserved.
//

import UIKit
import FFLabel

let demoContent = "#FFLabel#This is a @FFLabel Demo, access http://github.com/liufan321/fflabel can get the demo project. Follow @liufan2000 to get more information."

class ViewController: UIViewController, FFLabelDelegate {

    @IBOutlet weak var label: FFLabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        label.delegate = self
        label.text = demoContent
        label.font = UIFont.systemFont(ofSize: 20)
        label.textColor = UIColor.darkGray()
    }
    
    func labelDidSelectedLinkText(label: FFLabel, text: String) {
        print(text)
    }
}

